modules/fybrik-implicit-copy-batch/templates/batchtransfer.yaml

```diff
@@ -15,49 +15,56 @@ metadata:
 {{ end }}
 spec:
 {{- if and (.Values.assets) (eq (len .Values.assets) 1) -}}
-{{- with (index .Values.assets 0) }}
+{{- with $asset := (index .Values.assets 0) }}
+  {{- if and ($asset.args) (eq (len $asset.args) 2) -}}
+  {{- with index $asset.args 0 }}
   source:
-    {{ if .source.connection.s3 }}
+    {{- if index . "connection" "s3" }}
```

Comment:
Why not just .connection.s3 ?

Reply:
as far as I know its the same... anyway, I changed it.